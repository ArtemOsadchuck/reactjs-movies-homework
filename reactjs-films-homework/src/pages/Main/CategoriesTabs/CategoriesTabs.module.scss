$wite: rgb(255, 255, 255);
$black: rgb(0, 0, 0);

.categoriesWrapper {
  border: 1px solid $wite;
  border-radius: 5px;
  color: $wite;
  display: flex;
  font-size: 0.8rem;
  height: 2rem;
  position: absolute;
  right: 0.5rem;
  top: 0.5rem;
  transition: 0.3s ease-in-out;
  width: 20rem;

  .categoryBtn {
    align-items: center;
    display: flex;
    flex-grow: 1;
    height: 100%;
    justify-content: center;
    transition: 0.2s ease-in-out;
    user-select: none;

    &:hover {
      background-color: $wite;
      color: $black;
      cursor: pointer;
    }
  }

  .active {
    background-color: $wite;
    color: $black;

    &:hover {
      color: $black;
      cursor: inherit;
      opacity: 0.9;
    }
  }

  :nth-child(2) {
    border-left: 2px solid $wite;
    border-right: 2px solid $wite;
  }
}
