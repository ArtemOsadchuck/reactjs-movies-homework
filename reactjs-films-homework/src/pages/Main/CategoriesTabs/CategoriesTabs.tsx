import React, { useEffect } from 'react';
import { NavLink } from 'react-router-dom';

import styles from './CategoriesTabs.module.scss';

import lang from '../../../languages/getLanguage';
import { useAppSelector, useAppDispatch } from '../../../hooks/hooks';
import useUrlSearch from '../../../hooks/useUrlSearch';

import {
  setCategory,
  setActivePage,
} from '../../../store/rootStore/mainStore/mainSlice';

const CategoriesTabs: React.FC = () => {
  const appLang = useAppSelector((state) => state.mainReducer.lang);
  const dispatch = useAppDispatch();
  const activeClass = styles.categoryBtn + ' ' + styles.active;
  const btnClass = styles.categoryBtn;
  const activePage: string | undefined = useAppSelector(
    (state) => state.mainReducer.activePage
  );

  const startPage = '1';
  const activeUrlCategory = useUrlSearch('category');

  useEffect(() => {
    activePage && dispatch(setActivePage(activePage));
    activeUrlCategory && dispatch(setCategory(activeUrlCategory));
  }, [activePage, activeUrlCategory, dispatch]);

  const handlerSetCategory = (event: any) => {
    const category: string = event.target.id;
    dispatch(setActivePage(startPage));
    dispatch(setCategory(category));
  };

  return (
    <div className={styles.categoriesWrapper}>
      <NavLink
        to={`/?category=popular&page=${startPage}`}
        id={'popular'}
        onClick={(event) => handlerSetCategory(event)}
        className={activeUrlCategory === 'popular' ? activeClass : btnClass}
      >
        {lang(appLang).popular}
      </NavLink>
      <NavLink
        to={`?category=top_rated&page=${startPage}`}
        id={'top_rated'}
        onClick={(event) => handlerSetCategory(event)}
        className={activeUrlCategory === 'top_rated' ? activeClass : btnClass}
      >
        {lang(appLang).topRated}
      </NavLink>
      <NavLink
        to={`/?category=upcoming&page=${startPage}`}
        id={'upcoming'}
        onClick={(event) => handlerSetCategory(event)}
        className={activeUrlCategory === 'upcoming' ? activeClass : btnClass}
      >
        {lang(appLang).upcoming}
      </NavLink>
    </div>
  );
};

export default CategoriesTabs;
