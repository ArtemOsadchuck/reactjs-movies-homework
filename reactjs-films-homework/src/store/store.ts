import rootStore from './rootStore/rootStore';

export type RootState = ReturnType<typeof rootStore.getState>;
export type AppDispatch = typeof rootStore.dispatch;
export default rootStore;
