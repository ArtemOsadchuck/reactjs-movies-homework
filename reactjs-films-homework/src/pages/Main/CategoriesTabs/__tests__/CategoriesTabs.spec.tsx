import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';

import CategoriesTabs from '../CategoriesTabs';

import { Provider } from 'react-redux';
import store from '../../../../store/store';

describe('CategoriesTabs', () => {
  const history = createMemoryHistory();
  history.push('/');

  it('CategoriesTabs snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router history={history}>
          <CategoriesTabs />
        </Router>
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
    cleanup();
  });
});
