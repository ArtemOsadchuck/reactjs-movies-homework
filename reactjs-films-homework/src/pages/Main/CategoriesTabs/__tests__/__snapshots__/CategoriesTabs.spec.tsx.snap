// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesTabs CategoriesTabs snapshot 1`] = `
<DocumentFragment>
  <div
    class="categoriesWrapper"
  >
    <a
      aria-current="page"
      class="categoryBtn active"
      href="/?category=popular&page=1"
      id="popular"
    >
      Popular
    </a>
    <a
      aria-current="page"
      class="categoryBtn active"
      href="/?category=top_rated&page=1"
      id="top_rated"
    >
      Top rated
    </a>
    <a
      aria-current="page"
      class="categoryBtn active"
      href="/?category=upcoming&page=1"
      id="upcoming"
    >
      Upcoming
    </a>
  </div>
</DocumentFragment>
`;
